winml/test/model/compare_feature_value.cpp

```diff
@@ -0,0 +1,70 @@
+#include "testPch.h"
+#include "compare_feature_value.h"
+#include "StringHelpers.h"
+#include <core/framework/ml_value.h>
+#include "ort_value_helper.h"
+
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::AI::MachineLearning;
+
+namespace CompareFeatureValuesHelper{
+
+template <typename T>
+bool IsResultCloselyMatch(const T& outvalue, const T& expected_value, const double diff, const double tol) {
+  if (diff > tol) return false;
+  if (std::isnan(diff) && !(std::isnan(outvalue) && std::isnan(expected_value)) &&
+      !(std::isinf(outvalue) && std::isinf(expected_value)))
+    return false;
+  return true;
+}
```

Comment:
Should diff and tol be of type T?

Reply:
I followed this example here : https://github.com/microsoft/onnxruntime/blob/53307a5f2ea5a665a80fc55b40965c95f70bd1f6/onnxruntime/test/util/compare_ortvalue.cc#L59

perSampleTolerance and relativePerSampleTolerance are both always type double so I think it should continue to carry that datatype in calculations.